src/DssProxyActionsCharter.sol

```diff
@@ -0,0 +1,693 @@
+        }
+    }
+
+    function _getWipeDart(
+        address vat,
+        uint256 dai,
+        address urn,
+        bytes32 ilk
+    ) internal view returns (int256 dart) {
+        // Gets actual rate from the vat
+        (, uint256 rate,,,) = VatLike(vat).ilks(ilk);
+        // Gets actual art value of the urn
+        (, uint256 art) = VatLike(vat).urns(ilk, urn);
+
+        // Uses the whole dai balance in the vat to reduce the debt
+        dart = toInt256(dai / rate);
+        // Checks the calculated dart is not higher than urn.art (total debt), otherwise uses its value
+        dart = uint256(dart) <= art ? - dart : - toInt256(art);
+    }
+
+    function _getWipeAllWad(
+        address vat,
+        address urn,
+        bytes32 ilk
+    ) internal view returns (uint256 wad) {
+        // Gets actual rate from the vat
+        (, uint256 rate,,,) = VatLike(vat).ilks(ilk);
+        // Gets actual art value of the urn
+        (, uint256 art) = VatLike(vat).urns(ilk, urn);
+        // Gets actual dai amount in the urn
+        uint256 dai = VatLike(vat).dai(address(this));
+
+        uint256 rad = sub(mul(art, rate), dai);
+        wad = rad / RAY;
+
+        // If the rad precision has some dust, it will need to request for 1 extra wad wei
+        wad = mul(wad, RAY) < rad ? wad + 1 : wad;
+    }
+
+    // Public functions
+
+    function transfer(address gem, address dst, uint256 amt) public {
+        GemLike(gem).transfer(dst, amt);
+    }
+
+    function ethJoin_join(address charter, address apt) public payable {
+        // Wraps ETH in WETH
+        GemJoinLike(apt).gem().deposit{value: msg.value}();
+        // Approves adapter to take the WETH amount
+        GemJoinLike(apt).gem().approve(charter, msg.value);
+        // Joins WETH collateral into the vat
+        CharterLike(charter).join(apt, address(this), msg.value);
+    }
+
+    function gemJoin_join(address charter, address apt, uint256 amt) public {
+        // Gets token from the user's wallet
+        GemJoinLike(apt).gem().transferFrom(msg.sender, address(this), amt);
+        // Approves adapter to take the token amount
+        GemJoinLike(apt).gem().approve(charter, amt);
+        // Joins token collateral into the vat
+        CharterLike(charter).join(apt, address(this), amt);
+    }
+
+    function hope(
+        address obj,
+        address usr
+    ) public {
+        HopeLike(obj).hope(usr);
+    }
+
+    function nope(
+        address obj,
+        address usr
+    ) public {
+        HopeLike(obj).nope(usr);
+    }
+
+    function flux(
+        address charter,
+        bytes32 ilk,
+        address dst,
+        uint256 wad
+    ) public {
+        CharterLike(charter).flux(ilk, address(this), dst, wad);
+    }
+
+    function move(
+        address charter,
+        address dst,
+        uint256 rad
+    ) public {
+        VatLike(CharterLike(charter).vat()).move(address(this), dst, rad);
+    }
```

Comment:
I'm still figuring it out that we really need this one, but for now I'd just suggest to pass directly the `vat`, not really necessary to provide the charter.

Reply:
To be honest I don't think we need it at all.

Reply:
Hmm but in which part of the user flow they will do that?
I mean I see the users executing `lock`, `draw`, `wipe` and `free` ones. Why would they need to use this?

Reply:
Yes I agree.
Just that since we expose frob to the user we probably need this as well.
Otherwise we might want to make frob internal, and same for flux.
All of these (frob, move, flux) are exposed in the original proxy-actions though, so I followed the same rationale.

Reply:
As discussed, removed move and flux, made frob (now _frob) internal.